scripts/update-browser-releases/edge.ts

```diff
@@ -0,0 +1,271 @@
+
+import chalk from 'chalk-template';
+
+import { newBrowserEntry, updateBrowserEntry } from './utils.js';
+
+/**
+ * getFutureReleaseDate - Read the future release date
+ * @param {string} release The release number of the versin
+ * @param {string} releaseScheduleURL The url of the MD file having it
+ * @returns {string} The date in the YYYY-MM-DD format
+ */
+const getFutureReleaseDate = async (release, releaseScheduleURL) => {
+  // Fetch the MD of the release schedule
+  const scheduleMD = await fetch(releaseScheduleURL);
+  const text = await scheduleMD.text();
+  if (!text) {
+    console.error(chalk`{red Release file not found.}`);
+    return '';
+  }
+  // Find the line
+  //const regexp = new RegExp(`| ${release} |\\w*|\\w*| ?Week of (\\w*) ?|\\w*|`, 'i');
+  const regexp = new RegExp(
+    `\\| ${release} \\|.*\\|.*\\| ?Week of (.*) ?\\|.*\\|`,
+    'i',
+  );
+  const result = text.match(regexp);
+  if (!result) {
+    console.log(chalk`{yellow Release date not found for Edge ${release}.}`);
+    return '';
+  }
+  const releaseDateText = result[1];
+
+  // Get the date frome the file
+  const months = [
+    'Jan',
+    'Feb',
+    'Mar',
+    'Apr',
+    'May',
+    'Jun',
+    'Jul',
+    'Aug',
+    'Sep',
+    'Oct',
+    'Nov',
+    'Dec',
+  ];
+  const year = releaseDateText.substring(7, 11);
+  const month = months.indexOf(releaseDateText.substring(3, 6)) + 1;
+  const day = releaseDateText.substring(0, 2);
+
+  const releaseDate = new Date(`${year}-${month}-${day}Z`);
+
+  return releaseDate.toISOString().substring(0, 10); // Remove the time part
+};
+
+/**
+ * getReleaseNotesURL - Guess the URL of the release notes
+ * @param {string} status The status of the release
+ * @param {string} fullRelease The release of the release
+ * @param {string} date The date of the release
+ * @returns {string} The URL of the release notes or the empty string if not found
+ */
+const getReleaseNotesURL = async (status, fullRelease, date) => {
+  // If the status isn't stable, do not give back any release notes.
+  if (status !== 'Stable') {
+    return '';
+  }
+
+  const releaseStr = fullRelease.replace(/\./g, '');
+  const month = [
+    'january',
+    'february',
+    'march',
+    'april',
+    'may',
+    'june',
+    'july',
+    'august',
+    'september',
+    'october',
+    'november',
+    'december',
+  ];
+  const dateObj = new Date(date);
+  const dateStr = `${
+    month[dateObj.getMonth()]
+  }-${dateObj.getDate()}-${dateObj.getFullYear()}`;
+  return `https://learn.microsoft.com/en-us/deployedge/microsoft-edge-relnote-stable-channel#version-${releaseStr}-${dateStr}`;
+};
+
+/**
+ * updateEdgeReleases - Update the json file listing the browser releases of an Edge browser
+ * @param {object} options The list of options for Edge.
+ */
+export const updateEdgeReleases = async (options) => {
+  //
+  // Get the JSON with the versions from edgereleases
+  //
+  const buffer = await fetch(options.edgeupdatesURL);
```

Comment:
Minor bypass comment: not sure what the use case is for defining the URL in the index.ts file rather than as a cconst in the edge.ts instead. Seems like it's only used here and is very browser dependent, so might as well be hidden away in this file.

Reply:
I asked myself: if there is a second Edge (Edge for Android), like there is for Chrome or Firefox, what data will change? If this happens, only the `options` object in `index.ts` would need to be changed.